django_mongodb/query.py

```diff
@@ -120,6 +133,75 @@ def get_cursor(self):
         return self.collection.aggregate(pipeline)
 
 
+def join(self, compiler, connection):
+    lookups_pipeline = []
+    join_fields = self.join_fields or self.join_cols
+    lhs_fields = []
+    rhs_fields = []
+    for lhs, rhs in join_fields:
+        if isinstance(lhs, str):
+            lhs_mql = lhs
+            rhs_mql = rhs
+        else:
+            lhs, rhs = connection.ops.prepare_join_on_clause(
+                self.parent_alias, lhs, self.table_name, rhs
+            )
+            lhs_mql = lhs.as_mql(compiler, connection)
+            rhs_mql = rhs.as_mql(compiler, connection)
+            # replace prefix, in lookup stages the reference
+            # to this column is without the collection name.
+            rhs_mql = rhs_mql.replace(f"{self.table_name}.", "", 1)
+        lhs_fields.append(lhs_mql)
+        rhs_fields.append(rhs_mql)
+
+    parent_template = "parent__field__"
+    lookups_pipeline = [
+        {
+            "$lookup": {
+                "from": self.table_name,
+                "let": {
+                    f"{parent_template}{i}": parent_field
+                    for i, parent_field in enumerate(lhs_fields)
+                },
+                "pipeline": [
+                    {
+                        "$match": {
+                            "$expr": {
+                                "$and": [
+                                    {"$eq": [f"$${parent_template}{i}", field]}
+                                    for i, field in enumerate(rhs_fields)
+                                ]
+                            }
+                        }
+                    }
+                ],
+                "as": self.table_alias,
+            }
+        },
+    ]
+    if self.join_type != INNER:
```

Comment:
Can you add a comment about the purpose of this?

Reply:
Comments will be added.